collector/src/bin/rustc-perf-collector/execute.rs

```diff
@@ -262,6 +258,265 @@ lazy_static! {
     };
 }
 
+pub trait Processor {
```

Comment:
Please add some fairly short documentation describing what each method here is intended to do.